metricbeat/module/elasticsearch/index/data.go

```diff
@@ -59,27 +60,26 @@ func eventsMapping(r mb.ReporterV2, info elasticsearch.Info, content []byte) err
 	var indicesStruct IndicesStruct
 	err := json.Unmarshal(content, &indicesStruct)
 	if err != nil {
-		r.Error(err)
-		return err
+		return errors.Wrap(err, "failure parsing Elasticsearch Stats API response")
 	}
 
-	var errors multierror.Errors
+	var errs multierror.Errors
 	for name, index := range indicesStruct.Indices {
 		event := mb.Event{}
 		event.MetricSetFields, err = schema.Apply(index)
 		if err != nil {
-			r.Error(err)
-			errors = append(errors, err)
+			errs = append(errs, errors.Wrap(err, "failure applying index schema"))
```

Comment:
+1 to reporting 1 error per event, but only in the non-xpack code paths (since reporting errors into xpack indices doesn't make sense, at least today). I will fix all non-xpack code paths in this PR to go back to reporting 1 error per event.